src/ASM/ASMs3Dmx.h

```diff
@@ -199,6 +199,14 @@ class ASMs3Dmx : public ASMs3D, private ASMmxBase
   //! \param[out] n3 Number of nodes in third (w) direction
   //! \param[in] basis Which basis to return size parameters for
   virtual bool getSize(int& n1, int& n2, int& n3, int basis = 0) const;
+protected:
+  //! \brief Returns the volume in the parameter space for an element.
+  //! \param[in] iel Element index
+  double getParametricVolume(int iel) const;
+  //! \brief Returns boundary face area in the parameter space for an element.
+  //! \param[in] iel Element index
+  //! \param[in] dir Local face index of the boundary face
+  double getParametricArea(int iel, int dir) const;
```

Comment:
Which has more meaning than just cosmetics. But anyways my main question is why it should be virtual in the first place

-------- Opprinnelig melding --------
Fra: kmokstad <notifications@github.com>
Dato: 14.04.2016 17:51 (GMT+01:00)
Til: OPM/IFEM <IFEM@noreply.github.com>
Ko: Arne Morten Kvarving <Arne.Morten.Kvarving@sintef.no>
Emne: Re: [OPM/IFEM] added: method to add a global lagrange multiplier to the system (#12)


In src/ASM/ASMs3Dmx.h<https://github.com/OPM/IFEM/pull/12#discussion_r59741957>:

> @@ -199,6 +199,14 @@ class ASMs3Dmx : public ASMs3D, private ASMmxBase
>    //! \param[out] n3 Number of nodes in third (w) direction
>    //! \param[in] basis Which basis to return size parameters for
>    virtual bool getSize(int& n1, int& n2, int& n3, int basis = 0) const;
> +protected:
> +  //! \brief Returns the volume in the parameter space for an element.
> +  //! \param[in] iel Element index
> +  double getParametricVolume(int iel) const;
> +  //! \brief Returns boundary face area in the parameter space for an element.
> +  //! \param[in] iel Element index
> +  //! \param[in] dir Local face index of the boundary face
> +  double getParametricArea(int iel, int dir) const;


I know, but I prefer to declare virtual in the subclass also, such that it is clear that it is indeed virtual, without looking at the parent class definition.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/OPM/IFEM/pull/12/files/e4413e87fe4455ce9160c95fcc13c2455992a0c6..bff0ce9fa353fba985cba0d3c9cfa4ad48abed46#r59741957>